teps/0028-task-results-without-results.md

```diff
@@ -0,0 +1,358 @@
+skips that `pipelineTask`.
+
+### User Stories
+
+#### Story 1
+
+For a cleanup use case, `finally` task receives the project name, or the configuration name as a `task result` from
+the `acquire` task. If `acquire` task fails to acquire any resources, `cleanup` is skipped.
+
+#### Story 2
+
+Need user story for `finally` task with optional `param` referring to a task result.
+
+#### Story 3
+
+Need user story for a task with `continueAfterSkip` set to `true` with missing task result.
+
+#### Story 4
+
+Need user story for a when expression with missing task result.
+
+## Design Details
+
+### finally with missing task result
+
+`Pipeline` author can specify `default` for a missing result if that `param` is optional.
+
+```yaml
+spec:
+  params:
+    - name: owner-name
+      default: "the-best-owner"
+  tasks:
+   - name: boskos-acquire
+     taskRef:
+       name: boskos-acquire
+     params:
+       - name: owner-name
+         value: $(params.owner-name)
+     results:
+      - name: leased-resource
+        value: "fake-awesome-project"
+   - name: do-stuff-with-resource
+     taskRef:
+       name: do-stuff-with-resource
+     params:
+       - name: resource-name
+         value:  $(tasks.boskos-acquire.results.leased-resource)
+ finally:
+   - name: boskos-release
+     taskRef:
+       name: boskos-release
+     params:
+       - name: leased-resource
+         value:  $(tasks.boskos-acquire.results.leased-resource)
+```
+
+
+### when expression with missing task result
+
+```yaml
+spec:
+  tasks:
+    - name: add
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          value: 0
+    - name: multiply
+      when:
+        - input: "$(tasks.add.results.sum)"
+          operator: notin
+          values: ["0"]
+      taskRef:
+        name: multiply
+```
+
+### continueAfterSkip with missing task result
+
+```yaml
+spec:
+  tasks:
+    - name: add
+      when:
+        - input: "0"
+          operator: notin
+          values: ["0"]
+      taskRef:
+        name: add-task
+      params:
+        - name: first
+          value: $(params.first)
+        - name: second
+          value: $(params.second)
+      results:
+        - name: sum
+          value: 0
+    - name: multiply
+      runAfter: [ "add-task" ]
+      continueAfterSkip: "true"
+      taskRef:
+        name: multiply-by-99
+      params:
+        - name: x
+          value: $(task.add.results.sum)
+```
+
+
+## Test Plan
+
+e2e and unit tests
+
+## Alternatives
+
+### Option A
+
+A minor alternative to the proposal in which specifying `results` default values is limited to `optional` task `params`,
+instead, allow specifying default values for both `required` and `optional` task `params`. And skip the task when
+default values are not specified while results are missing.
```

Comment:
What would be the reason to not allow the default result value to be used with a required parameter?  The pipeline author might be trying to reuse a task that does not have a default for a param value.  It seems to me like the pipeline author should be able to pass the default result value anyway.

Reply:
It was adhering to the same concept of required v/s optional params. Dropped this limitation.